apps/jobs/views.py

```diff
@@ -360,7 +360,14 @@ def leaderboard(request, challenge_phase_split_id):
 
     leaderboard_labels = challenge_phase_split.leaderboard.schema['labels']
     for item in distinct_sorted_leaderboard_data:
-        item['result'] = [item['result'][index] for index in leaderboard_labels]
+        result = []
```

Comment:
@Ram81 Another better way of doing it to add this check while entering the data in the LeaderboardData Table i.e. in submission worker script.
 So, the flow would be
1. Check for the metrics in the particular leaderboard corresponding to the submission result.
2. If the keys match then the submission will be successful and data will be entered in the LeaderboardData table.
3. If the keys don't match then submission status will be set to failed, the error will be stored in the std_error file, LeaderboardData object won't be create and execution will be terminated.

Please remove this code and the corresponding test cases for it. We can merge the remaining test cases that you have added.

cc: @deshraj

Reply:
Sure